src/Lean/Meta/Tactic/Rewrite.lean

```diff
@@ -45,8 +45,11 @@ def rewrite (mvarId : MVarId) (e : Expr) (heq : Expr)
           let eqRefl ← mkEqRefl e
           let eqPrf ← mkEqNDRec motive eqRefl heq
           postprocessAppMVars `rewrite mvarId newMVars binderInfos
-          let newMVars ← newMVars.filterM fun mvar => not <$> isExprMVarAssigned mvar.mvarId!
-          pure { eNew := eNew, eqProof := eqPrf, mvarIds := newMVars.toList.map Expr.mvarId! }
+          let newMVarIds ← newMVars.map Expr.mvarId! |>.filterM (not <$> isExprMVarAssigned ·)
+          let otherMVarIds ← getMVarsNoDelayed eqPrf
```

Comment:
@leodemoura We might want to support all `ApplyNewGoals` modes, once they are implemented for `apply`, for `rw` as well, but for now I just wanted to make sure we don't lose track of any unsolved mvars